### PR TITLE
Add 1 minute time difference check

### PR DESCRIPTION
**Pull request by kevinchan159**
Issue #149 

- Ensure that routes always have a minimum one minute difference

Verified on dev server

Screenshots:
52444227_2467266743501327_2349293887677267968_n
51666596_347948599137613_590690661854871552_n

### CODE REVIEW

File: src/utils/ParseRouteUtils.js [hunk 1]
```diff
@@ -222,9 +223,36 @@ async function condenseRoute(
...
+        ).toISOString().split('.')[0]}Z`;
+    }
+}
+
+/**
+ * Returns whether departureDate and arrivalDate are within the same minute
+ * @param route
+ * @returns boolean
+ */
+function isValidDates(departureDate: Date, arrivalDate: Date): boolean {
```
Comment (young-k) [inline src/utils/ParseRouteUtils.js]: this function isn't named properly, `isValidDates`? it should describe what the function is doing
also `@param route` isn't correct either

File: src/utils/ParseRouteUtils.js [hunk 2]
```diff
@@ -222,9 +223,36 @@ async function condenseRoute(
...
+    return !(departureDate.getTime() + ONE_MIN_IN_MILLISECONDS > arrivalDate.getTime()
```
Comment (young-k) [inline src/utils/ParseRouteUtils.js]: why is the first check necessary?

Comment (kevinchan159) [reply src/utils/ParseRouteUtils.js]: the minutes could be the same but the hour is different

Comment (young-k) [reply src/utils/ParseRouteUtils.js]: ah you are smart, can we make these explicit as in:
```
const isSameMinute = ...
const isSameHour = ...
return isSameMinute && isSameHour;
```
it'll be easier to understand in the future

**Comment by young-k** [comment]
In the future, you should run `git commit --amend`, and `git push --force`, rather than making a new commit to address the feedback!

File: src/utils/ParseRouteUtils.js [hunk 3]
```diff
@@ -222,9 +223,38 @@ async function condenseRoute(
...
+    }
+}
+
+/**
+ * Returns whether departureDate and arrivalDate are within the same minute
+ * @param departureDate
+ * @param arrivalDate
+ * @returns boolean
+ */
+function datesAreSameMinute(departureDate: Date, arrivalDate: Date): boolean {
```
Comment (young-k) [inline src/utils/ParseRouteUtils.js]: maybe `areDatesTheSameMinute`? the more verbose the better

**Comment by kevinchan159** [comment]
> In the future, you should run `git commit --amend`, and `git push --force`, rather than making a new commit to address the feedback!

I was gonna squash all of them after @alanna-zhou reviews it in case I need to change anything else

**Pull request merged**